### DISCUSSION TITLE
Choose plan9 user for wsl$ network share

### DISCUSSION THREAD

**Discussion by edrevo** [microsoft/WSL#5210] [General]
**Is your feature request related to a problem? Please describe.**
I am mounting my Ubuntu distro as a network share through \\wsl$, but any files created in Windows through the share appear as root:root in Linux. Is there any way to change this? Perhaps there is some way of configuring the P9 server in WSL2? I have searched for documentation but I can't find anything.

**Describe the solution you'd like**
Being able to make changes to the WSL2 file system with a non-root user.

**Describe alternatives you've considered**
sudo chown constantly, but that's a pain

**Comment by therealkenc** [comment]
Imported distro, right?

Start by trying #3974 ([message](https://github.com/microsoft/WSL/issues/3974#issuecomment-576782860)). If that doesn't take, default-user-id it harder in the registry ([message](https://github.com/microsoft/WSL/issues/3974#issuecomment-522921145)). Either way do a `wsl.exe --shutdown` (or a reboot) before accessing 9p in either direction after setting the default user id.

**Comment by kitor** [comment]
Hi, 
I'd like to bump this. I have a setup where I need my default user to stay as root, however would like to map files in wsl share to different unix user. I think this functionality could be easily extended by adding additional option to wsl.conf that would just fallback to default user if not set.

Is there any workaround I can use?

For the record, it's WSL1 with Debian image from store. Nothing fancy.
